src/Domain/Creators/TimeDependence/CubicScale.hpp

```diff
@@ -59,6 +59,12 @@ class CubicScale final : public TimeDependence<MeshDim> {
     static constexpr Options::String help = {
         "The initial time of the functions of time"};
   };
+  /// \brief The time interval for updates of the functions of time.
+  struct UpdateDeltaT {
+    using type = double;
+    static constexpr Options::String help = {
+        "The time interval for updates of the functions of time"};
```

Comment:
I think you should add that `If the function or expiration time is not updated by the end of the interval the simulation will fail.` because you haven't really supplied any way of saying "It never gets updated." Would this be a good place for using `Auto`?

Reply:
Yes, I agree that UpdateDeltaT is not optimal, partially because there's no way of saying "It never gets updated". Furthermore, I don't like the need to specify UpdateDeltaT here at all, because expiration times should be set by the control system rather than by the DomainCreator constructors.   However, the `FunctionsOfTime::PiecewisePolynomial` needs an expiration time in its constructor, and that constructor  is called by `CubicScale::functions_of_time()`.  Note that `CubicScale::functions_of_time()` is the only place `UpdateDeltaT` is used.

I can see a few possibilities to improve this situation:

 - Rename `UpdateDeltaT` to `InitialExpirationTime` since the option is used only for initializing the `PiecewisePolynomial`. Then use `Options::Auto` to set the automatic value to  `std::numeric_limits<double>::max()` (a.k.a. "it never gets updated"). Rely on the control system to reset the expiration time as needed.
 - Eliminate the option entirely, and set the initial expiration time to `std::numeric_limits<double>::max()` always, expecting that the control system will reset the expiration time as needed.
 
Both of the above  possibilities (as stated) would require me to remove the `next_expiration_time >= expiration_time_` check in `PiecewisePolynomial::reset_expiration_time` (because the initial expiration time is `std::numeric_limits<double>::max`, so the check would always fail).  If we still want the check in `PiecewisePolynomial::reset_expiration_time`, we could do either:
 - Set the initial expiration time to `std::numeric_limits<double>::lowest()` so that the `PiecewisePolynomial` is initially invalid, and then rely on the control system to reset it, including before the first time it gets used.
 - Set the initial expiration time to `initial_time`, so the `PiecewisePolynomial` is initially valid only for its initial time.  Rely on the control system to reset it before it is used for any subsequent times.
 - Change `PiecewisePolynomial::expiration_time_` from a double to a `std::optional<double>`, initially default-constructed, and interpret an empty `expiration_time_` as "never expires". Allow `PiecewisePolynomial::reset_expiration_time` to reset an empty `expiration_time_` to anything.

Thoughts?

Reply:
I currently favor eliminating the option entirely, because I don't know how the DomainCreator is going to know about the initial expiration time.

Reply:
Consensus from SpECTRE meeting Nov 23, 2020: Rename `UpdateDeltaT` -> `InitialExpirationDeltaT`, add `Auto` to this option so that `Auto` makes `expiration_time` become `std::numeric_limits<double>::max()` (i.e. never expires), which is never resettable or updatable.